### PR TITLE
Add `Efficientdet-B0` implementation

### PR DESCRIPTION
**Pull request by willthefrog**
see title

### CODE REVIEW

File: ppdet/data/transform/operators.py [hunk 1]
```diff
@@ -1672,3 +1673,217 @@ def __call__(self, sample, context=None):
...
+        interp = self.interp
+        dim = self.target_dim
+        dim_max = max(h, w)
+        scale = self.target_dim / dim_max
+        resize_w = int(round(w * scale))
+        resize_h = int(round(h * scale))
+        if 'gt_bbox' in sample and len(sample['gt_bbox']) > 0:
+            scale_array = np.array([scale, scale] * 2, dtype=np.float32)
+            sample['gt_bbox'] = np.clip(sample['gt_bbox'] * scale_array, 0,
+                                        dim - 1)
```
Comment (jerrywgz) [inline ppdet/data/transform/operators.py:1778]: Resize部分和class Resize是相同的逻辑，能否复用？

File: ppdet/modeling/anchor_heads/efficient_head.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
+# Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
```
Comment (jerrywgz) [inline ppdet/modeling/anchor_heads/efficient_head.py:1]: 2020

File: ppdet/modeling/architectures/efficientdet.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            use_dataloader=True,
+            iterable=False):
+        inputs_def = self._inputs_def(image_shape)
+        feed_vars = OrderedDict([(key, fluid.data(
+            name=key,
+            shape=inputs_def[key]['shape'],
+            dtype=inputs_def[key]['dtype'])) for key in fields])
+        loader = fluid.io.DataLoader.from_generator(
+            feed_list=list(feed_vars.values()),
+            capacity=64,
```
Comment (jerrywgz) [inline ppdet/modeling/architectures/efficientdet.py:112]: 目前capacity统一为16

File: ppdet/data/transform/operators.py [hunk 4]
```diff
@@ -1672,3 +1673,217 @@ def __call__(self, sample, context=None):
         bbox[:, :2] = bbox[:, :2] + bbox[:, 2:4] / 2.
         sample['gt_bbox'] = bbox
         return sample
+
+
+@register_op
+class FilterCrowd(BaseOperator):
```
[the remark this answers is not in the document]
Comment (willthefrog) [reply ppdet/data/transform/operators.py:1679]: 默认就是false， 那可以去掉了。

[the remark this answers is not in the document]
Comment (willthefrog) [reply ppdet/data/transform/operators.py:1679]: done

File: ppdet/data/transform/operators.py (hunk 1 above)
Comment (willthefrog) [reply ppdet/data/transform/operators.py:1778]: 不一致，这里是按照长边resize再pad。

File: ppdet/modeling/anchor_heads/efficient_head.py (hunk 2 above)
Comment (willthefrog) [reply ppdet/modeling/anchor_heads/efficient_head.py:1]: done

File: ppdet/modeling/architectures/efficientdet.py (hunk 3 above)
Comment (willthefrog) [reply ppdet/modeling/architectures/efficientdet.py:112]: done

File: configs/efficientdet_d0.yml [hunk 5]
```diff
@@ -0,0 +1,157 @@
+architecture: EfficientDet
```
Comment (jerrywgz) [inline configs/efficientdet_d0.yml:1]: 需要增加下文档并上传模型

File: tools/train.py [hunk 6]
```diff
@@ -127,10 +129,21 @@ def main():
                     loss *= ctx.get_loss_scale_var()
                 lr = lr_builder()
                 optimizer = optim_builder(lr)
-                optimizer.minimize(loss)
+                clip = None
+                if optim_builder.clip_grad_by_norm is not None:
+                    clip = fluid.clip.GradientClipByGlobalNorm(
+                        clip_norm=optim_builder.clip_grad_by_norm)
+                optimizer.minimize(loss, grad_clip=clip)
```
Comment (qingqing01) [inline tools/train.py:136]: In develop, grad_clip is moved to init in optimizer

File: configs/efficientdet_d0.yml (hunk 5 above)
Comment (willthefrog) [reply configs/efficientdet_d0.yml:1]: done.

File: tools/train.py (hunk 6 above)
Comment (willthefrog) [reply tools/train.py:136]: will this change land in 2.0?

File: configs/efficientdet_d0.yml (hunk 5 above)
Comment (willthefrog) [reply configs/efficientdet_d0.yml:1]: done

**Comment by willthefrog** [comment]
> There is conflicts.

resolved

**Comment by willthefrog** [comment]
> Still has conflicts

rebased

**Pull request merged**